build.sbt

```diff
@@ -82,31 +83,37 @@ val plugin = project.settings(
         s"scala3-compiler_3"
       else "scala-compiler"
     ) % scalaVersion.value
-  )
+  ),
+  Compile / unmanagedSourceDirectories ++= {
+    val extraDirectoriesWithPredicates = Map[String, String => Boolean](
```

Comment:
Good enough for now, but in the future I suspect the compiler phases may be split/merged in new ways too - we'll need to support more version ranges if that happens.

For now the only new split is "before 3.1.0 milestones" and "everything after that".